File: public/extract.py
#!/usr/bin/env python3
import re
import os

# Read the HTML file
with open('index.html', 'r', encoding='utf-8') as f:
    content = f.read()

# Extract CSS (between <style> and </style>)
css_match = re.search(r'<style>(.*?)</style>', content, re.DOTALL)
if css_match:
    css_content = css_match.group(1)
    os.makedirs('css', exist_ok=True)
    with open('css/styles.css', 'w', encoding='utf-8') as f:
        f.write(css_content)
    print(f'✅ CSS extracted: {len(css_content)} characters')

# Extract JavaScript (between <script> and </script>, but skip Firebase script tags)
# Find the main script block (after Firebase scripts)
script_start = content.find('<script>', content.find('</style>'))
script_end = content.rfind('</script>')
if script_start != -1 and script_end != -1:
    js_content = content[script_start + 8:script_end]
    os.makedirs('js', exist_ok=True)
    with open('js/app.js', 'w', encoding='utf-8') as f:
        f.write(js_content)
    print(f'✅ JavaScript extracted: {len(js_content)} characters')

# Update HTML - replace <style> with <link> and <script> with external reference
updated_html = content

# Replace <style>...</style> with <link>
updated_html = re.sub(r'<style>.*?</style>', '<link rel="stylesheet" href="/css/styles.css">', updated_html, flags=re.DOTALL)

# Replace main <script> with external reference (but keep Firebase scripts)
# Find the main script block
main_script_start = content.find('<script>', content.find('</style>'))
main_script_end = content.rfind('</script>')
if main_script_start != -1 and main_script_end != -1:
    before_script = updated_html[:main_script_start]
    after_script = updated_html[main_script_end + 9:]
    updated_html = before_script + '<script src="/js/app.js"></script>' + after_script

# Write updated HTML
with open('index.html', 'w', encoding='utf-8') as f:
    f.write(updated_html)
print('✅ HTML updated with external CSS and JS references')
print('✅ Modular transformation complete!')

File: public/extract.js
const fs = require('fs');
const path = require('path');

const filePath = path.join(__dirname, 'index.html');
console.log('Reading from:', filePath);

if (!fs.existsSync(filePath)) {
  console.error('File not found:', filePath);
  process.exit(1);
}

const content = fs.readFileSync(filePath, 'utf8');
console.log('File read, length:', content.length);

// Extract CSS
const cssMatch = content.match(/<style>([\s\S]*?)<\/style>/);
if (!cssMatch) {
  console.error('CSS not found in file');
  process.exit(1);
}
const css = cssMatch[1];

// Create css directory if it doesn't exist
const cssDir = path.join(__dirname, 'css');
if (!fs.existsSync(cssDir)) {
  fs.mkdirSync(cssDir, { recursive: true });
}

// Write CSS file
fs.writeFileSync(path.join(cssDir, 'styles.css'), css, 'utf8');
console.log('CSS extracted:', css.length, 'characters');

// Extract JavaScript
const scriptStart = content.indexOf('<script>', content.indexOf('</style>')) + 8;
const scriptEnd = content.lastIndexOf('</script>');
const js = content.substring(scriptStart, scriptEnd);

// Create js directory if it doesn't exist
const jsDir = path.join(__dirname, 'js');
if (!fs.existsSync(jsDir)) {
  fs.mkdirSync(jsDir, { recursive: true });
}

// Write JavaScript file
fs.writeFileSync(path.join(jsDir, 'app.js'), js, 'utf8');
console.log('✅ JavaScript extracted:', js.length, 'characters');

// Update HTML - replace <style>...</style> with <link>
let updatedHtml = content;
updatedHtml = updatedHtml.replace(/<style>[\s\S]*?<\/style>/, '<link rel="stylesheet" href="/css/styles.css">');
console.log('✅ Replaced <style> tag with <link>');

// Replace main <script> with external reference (but keep Firebase scripts)
const mainScriptStart = updatedHtml.indexOf('<script>', updatedHtml.indexOf('</style>'));
const mainScriptEnd = updatedHtml.lastIndexOf('</script>');
if (mainScriptStart !== -1 && mainScriptEnd !== -1) {
  const beforeScript = updatedHtml.substring(0, mainScriptStart);
  const afterScript = updatedHtml.substring(mainScriptEnd + 9);
  updatedHtml = beforeScript + '<script src="/js/app.js"></script>' + afterScript;
  console.log('✅ Replaced main <script> tag with <script src>');
}

fs.writeFileSync(filePath, updatedHtml, 'utf8');
console.log('✅ HTML updated with external CSS and JS references');
console.log('✅ Modular transformation complete!');

